chatkit-android/src/main/kotlin/com/pusher/chatkit/BeamsPushNotifications.kt

```diff
@@ -0,0 +1,77 @@
+package com.pusher.chatkit
+
+import android.content.Context
+import com.pusher.chatkit.pushnotifications.BeamsTokenProviderService
+import com.pusher.chatkit.pushnotifications.PushNotifications
+import com.pusher.pushnotifications.Callback
+import com.pusher.pushnotifications.PusherCallbackError
+import com.pusher.pushnotifications.auth.TokenProvider
+import com.pusher.util.Result
+import elements.Error
+import java8.util.concurrent.CompletableFuture
```

Comment:
Is this some kind of shim for pre-java8 support?

The rest of the library uses Pablo's simpler re-implementation, which should be sufficient for what you're doing here without additional dependencies.